dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java

```diff
@@ -485,6 +485,30 @@ public void testGetEnrollmentUserWithNoDataReadWriteOrgUnit()
         enrollmentService.getEnrollment( importSummary.getReference() );
     }
 
+    @Test
+    public void testAddEnrollmentToOrgUnitWithoutProgramAccess()
+    {
+        programA.getSharing().setPublicAccess( AccessStringHelper.DEFAULT );
+        manager.updateNoAcl( programA );
+        Enrollment en = createEnrollment( programA.getUid(), maleA.getUid() );
+        en.setOrgUnit( organisationUnitB.getUid() );
+        ImportSummary importSummary = enrollmentService.addEnrollment(
+            en, ImportOptions.getDefaultImportOptions() );
+
+        assertEquals( ImportStatus.ERROR, importSummary.getStatus() );
+        assertEquals( "Program is not assigned to this Organisation Unit: " + organisationUnitB.getUid(),
+            importSummary.getDescription() );
```

Comment:
in my opinion we shouldn't test error messages, i don't see a good reason because messages could change over time, there could be localizations, etc so we should just test  a status or better an error code

Reply:
I get your point. In this case there is no error code in the old import Summary and the test had to make sure it was the specific validation that failed instead of something else. 
If the error message changes, this test would fail and that means this test has to be updated. 
At the moment, there is no localisation for these importSummaries, atleast not in tests. 🤔

Reply:
yes it is a general overview of the problem, it doesn't have to be this particular case, testing specific messages it is not a good practice in my opinion